test/data/javascript.js

```diff
@@ -92,6 +94,800 @@ exports.test_data = {
+                input: inputlib.operator_position.colon_special_case,
+                output: [
+                    'var a = {',
+                    '    b: bval,',
+                    '    c: cval',
+                    '};',
+                    'var d = e ? f : g;'
+                ]
+            }, {
+                comment: 'catch-all, includes brackets and other various code',
+                unchanged: inputlib.operator_position.catch_all
+            }
+        ]
+    }, {
+        name: "operator_position option - set to preserve_newline",
+        options: [
+            { name: "operator_position", value: "'remove_newline'" },
+            { name: "preserve_newlines", value: "false" },
+        ],
+        tests: [
+            {
+                comment: 'comprehensive, but simple',
+                unchanged: inputlib.operator_position.comprehensive_simple
+            }, {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive_newlines,
+                output: [
+                    'var res = a + b - c / d * e % f;',
+                    'var res = g & h | i ^ j;',
+                    'var res = (k && l || m) ? n : o;',
+                    'var res = p >> q << r >>> s;',
+                    'var res = t === u !== v != w == x >= y <= z > aa < bb;'
+                ]
+            }, {
+                comment: 'handle colon special case',
+                input: inputlib.operator_position.colon_special_case,
+                output: [
+                    'var a = {',
+                    '    b: bval,',
+                    '    c: cval',
+                    '};',
+                    'var d = e ? f : g;'
+                ]
+            }, {
+                comment: 'catch-all, includes brackets and other various code',
+                input: inputlib.operator_position.catch_all,
+                output: [
+                    'var d = 1;',
+                    'if (a === b && c) {',
+                    '    d = (c * everything / something_else) % b;',
+                    '    e += d;',
+                    '} else if (!(complex && simple) || (emotion && emotion.name === "happy")) {',
+                    '    cryTearsOfJoy(many || anOcean || aRiver);',
+                    '}'
+                ]
+            }
+        ]
+    }, {
+        name: "operator_position option - set to preserve_newline",
+        options: [
+            { name: "operator_position", value: "'remove_newline'" },
+            { name: "preserve_newlines", value: "true" },
+        ],
+        tests: [
+            {
+                comment: 'comprehensive, but simple',
+                unchanged: inputlib.operator_position.comprehensive_simple
+            }, {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive_newlines,
+                output: [
+                    'var res = a + b - c /',
```

Comment:
This seems wrong.  The default is not `remove_newline`, so if someone sets `operator_position  = remove_newline`, it should take precedence over the general `preserve_newlines = true` setting.   That is what you have it doing for `operator_position  = preserve_newline` with ``preserve_newlines = false`, so the inverse should also hold.